### ISSUE TITLE
DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats

### ISSUE THREAD

**Issue by mgedmin** [python-pillow/Pillow#3750]
### What did you do?

Ran my test suite on Python 3.8.0a3

### What did you expect to happen?

The test to pass

### What actually happened?

The tests did pass, but there were new deprecation warnings from Pillow:

```
.../.tox/py38/lib/python3.8/site-packages/PIL/Image.py:477: DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats
    return encoder(mode, *args + extra)
```

These were introduced by https://bugs.python.org/issue36381 to warn about an upcoming Python C API change. The meaning of PY_SSIZE_T_CLEAN is described in https://python.readthedocs.io/en/stable/c-api/arg.html#strings-and-buffers.

What Pillow needs to do is:

- find all invocations of PyArg_Parse and related functions that use # formats
make sure the type of the length argument is a Py_ssize_t
- #define PY_SSIZE_T_CLEAN above the #include <Python.h>.

PY_SSIZE_T_CLEAN was introduced in Python 2.5 so there shouldn't be any backwards-compatibility issues.

### What are your OS, Python and Pillow versions?

* OS: Ubuntu 18.10
* Python: 3.8.0a3
* Pillow: 5.4.1

### Reproduction

```python
#!/usr/bin/env python3.8
import warnings
warnings.simplefilter("once")
from PIL import Image
Image.new("RGB", (1, 1)).save("/tmp/pixel.png")
```

**Comment by hugovk** [comment]
Please could you test with https://github.com/python-pillow/Pillow/pull/3749?

**Comment by mgedmin** [comment]
Of course I did a search for existing _issues_ and didn't think to check for existing _pull requests_.

It looks like the same issue, and the PR looks good to me (who is an absolute stranger to the Python C API).

How do I pip install a github PR into a virtualenv to test?

**Comment by mgedmin** [comment]
> How do I pip install a github PR into a virtualenv to test?

What I did was clone the repo, fetch the pull/3749 ref, check it out, and `.tox/py36/bin/pip install ~/src/Pillow`.

The deprecation warning is gone and all my tests pass.

**Comment by hugovk** [comment]
Thanks for the report and testing, https://github.com/python-pillow/Pillow/pull/3749 has been merged and will be in Monday's 6.0.0 release.
